lib/ansible/modules/cloud/azure/azure_rm_batchaccount.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2019 Junyi Yi (@JunyiYi)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_batchaccount
+version_added: "2.9"
+
+short_description: Manages a Batch Account on Azure.
+
+description:
+    - Create, update and delete instance of Azure Batch Account.
+
+options:
+    resource_group:
+        description:
+        - The name of the resource group in which to create the Batch Account.
+        required: true
+    name:
+        description:
+        - The name of the Batch Account.
+        required: true
+    location:
+        description:
+        - Specifies the supported Azure location where the resource exists.
+        required: true
+    auto_storage_account_id:
+        description:
+        - The ID of the Batch Account auto storage account.
+        required: false
+    key_vault_reference:
+        description:
+        - A reference to the Azure key vault associated with the Batch account.
+        required: false
+        suboptions:
+            id:
+                description:
+                - The resource ID of the Azure key vault associated with the Batch
+                    account.
+                required: true
+            url:
+                description:
+                - The URL of the Azure key vault associated with the Batch account.
+                required: true
+    pool_allocation_mode:
+        description:
+        - The pool acclocation mode of the Batch Account.
+        required: false
+        default: BatchService
+        choices:
+        - BatchService
+        - UserSubscription
+    tags:
```

Comment:
that should be removed and will be added by common azure_tags documentation fragment